src/communication/gathered_vector.hpp

```diff
@@ -28,6 +30,15 @@ class gathered_vector {
         return partition_;
     }
 
+    // iterators to a distribution partition
```

Comment:
The unit test only covers the low level `communication::global_policy::gather_spikes()` method which takes the vector of local spikes, and does whatever communication is required by the global policy to make the global spike list.

We need some generic unit tests to check whether these different `make_event_queues` methods generate the right events in queues given a set of global spikes.